lib/looker-sdk/client.rb

```diff
@@ -296,12 +304,16 @@ def reset_agent
       @agent = nil
     end
 
-    def request(method, path, data, options, &block)
+    def request(method, path, data, options, encoded, &block)
       ensure_logged_in
       begin
+        path = path.to_s
+        if !encoded
+          path = URI::Parser.new.escape(path)
+        end
```

Comment:
this is a bugfix from an earlier patch I made to auto-urlencode path params. That change broke the case of calling `get/post/put/patch` directly with a path that needed encoding. Unfortunately the "auto-encoding" at this level doesn't do the right thing so we need this extra `encoded` param so the dynamic client can call with `encoded=true` but direct http calls get encoded